src/NuGet.Core/NuGet.Protocol/Plugins/Logging/IPluginLogger.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Protocol.Plugins
+{
+    public interface IPluginLogger : IDisposable
```

Comment:
Why does this need to be public?

Reply:
If it's for test, i think it's fine to use InternalsVisibleTo similar to https://github.com/NuGet/NuGet.Client/commit/1fe979bfbfcd1debf492d27446c3474c42a58c5b#diff-aa45b664821154d7562b7b6575d05bc6

Reply:
Good point.  I made it public because I added it as a parameter on some public constructors, but I can make those as well as this `internal`.